File: backend/app/schemas/study_session.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class StudySessionCreate(BaseModel):
    duration_minutes: int
    notes_reviewed: Optional[int] = 0
    flashcards_practiced: Optional[int] = 0
    subject: Optional[str] = None
    session_type: Optional[str] = None

class StudySessionResponse(StudySessionCreate):
    id: int
    user_id: int
    created_at: datetime

    class Config:
        from_attributes = True

class StudySessionListResponse(BaseModel):
    sessions: list[StudySessionResponse]
    total: int

File: frontend/lib/models/study_session.dart
class StudySession {
  final int id;
  final int userId;
  final int durationMinutes;
  final int notesReviewed;
  final int flashcardsPraticed;
  final String? subject;
  final String? sessionType;
  final DateTime createdAt;

  StudySession({
    required this.id,
    required this.userId,
    required this.durationMinutes,
    required this.notesReviewed,
    required this.flashcardsPraticed,
    this.subject,
    this.sessionType,
    required this.createdAt,
  });

  factory StudySession.fromJson(Map<String, dynamic> json) {
    return StudySession(
      id: json['id'],
      userId: json['user_id'],
      durationMinutes: json['duration_minutes'],
      notesReviewed: json['notes_reviewed'],
      flashcardsPraticed: json['flashcards_practiced'],
      subject: json['subject'],
      sessionType: json['session_type'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }
}
